Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/EditorAction.cpp

```diff
@@ -34,15 +36,15 @@ namespace AzToolsFramework
             }
         );
 
-        if (updateCallback)
+        if (checkStateCallback)
         {
             m_action->setCheckable(true);
-            m_updateCallback = AZStd::move(updateCallback);
+            m_checkStateCallback = AZStd::move(checkStateCallback);
```

Comment:
is this necessary? in RegisterCheckableAction::updateCallback and handle are passed by value as well. you can pass by reference and do a straight assignment.. any reason to consume the function like this. seems like something that could cause problems if